server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -879,22 +878,6 @@ public void run() {
       }
     }
 
-    // wait for shutdown
-    // if the main thread exits oldServer the manager listener, the JVM will
-    // kill the other threads and finalize objects. We want the shutdown that is
-    // running in the manager listener thread to complete oldServer this happens.
-    // consider making other threads daemon threads so that objects don't
-    // get prematurely finalized
-    synchronized (this) {
-      while (!shutdownComplete) {
-        try {
-          this.wait(1000);
-        } catch (InterruptedException e) {
-          log.error(e.toString());
-        }
-      }
-    }
```

Comment:
Certainly if `shutdownComplete` is not set, this would cause the tserver to not shutdown normally I think.